examples/src/main/java/io/dapr/examples/tracing/DemoServiceController.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+package io.dapr.examples.tracing;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.opentelemetry.trace.Tracer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * SpringBoot Controller to handle service invocation.
+ *
+ * <p>Instrumentation is handled in {@link io.dapr.springboot.OpenTelemetryInterceptor}.
+ */
+@RestController
+public class DemoServiceController {

```

Comment:
DemoServiceController is defined in both invoke sample and tracing sample. This leads to exception when the sample is launched:

←[94;1m== APP == 2020-08-17 12:56:44,930 {HH:mm:ss.SSS} [main] INFO  io.dapr.examples.tracing.DemoService - Starting DemoService on DESKTOP-HRKK1LN with PID 19900 (C:\Users\hbai\go\src\github.com\dapr\java-sdk\examples\target\dapr-java-sdk-examples-exec.jar started by hbai in C:\Users\hbai\go\src\github.com\dapr\java-sdk\examples)
←[0m
←[94;1m== APP == 2020-08-17 12:56:44,932 {HH:mm:ss.SSS} [main] INFO  io.dapr.examples.tracing.DemoService - No active profile set, falling back to default profiles: default
←[0m
←[94;1m== APP == 2020-08-17 12:56:45,386 {HH:mm:ss.SSS} [main] WARN  o.s.b.w.s.c.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [io.dapr.springboot.DaprApplication]; nested exception is org.springframework.context.annotation.ConflictingBeanDefinitionException: Annotation-specified bean name 'demoServiceController' for bean class [io.dapr.examples.tracing.DemoServiceController] conflicts with existing, non-compatible bean definition of same name and class [io.dapr.examples.invoke.http.DemoServiceController]
←[0m
←[94;1m== APP == 2020-08-17 12:56:45,724 {HH:mm:ss.SSS} [background-preinit] WARN  o.s.h.c.j.Jackson2ObjectMapperBuilder - For Jackson Kotlin classes support please add "com.fasterxml.jackson.module:jackson-module-kotlin" to the classpath
←[0m
←[94;1m== APP == 2020-08-17 12:56:45,753 {HH:mm:ss.SSS} [main] ERROR o.s.boot.SpringApplication - Application run failed
←[0m
←[94;1m== APP == org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [io.dapr.springboot.DaprApplication]; nested exception is org.springframework.context.annotation.ConflictingBeanDefinitionException: Annotation-specified bean name 'demoServiceController' for bean class [io.dapr.examples.tracing.DemoServiceController] conflicts with existing, non-compatible bean definition of same name and class [io.dapr.examples.invoke.http.DemoServiceController]
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:184)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:325)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:242)
←[0m
←[94;1m== APP ==        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:275)
←[0m
←[94;1m== APP ==        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:95)
←[0m
←[94;1m== APP ==        at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:706)
←[0m
←[94;1m== APP ==        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:532)
←[0m
←[94;1m== APP ==        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141)
←[0m
←[94;1m== APP ==        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747)
←[0m
←[94;1m== APP ==        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
←[0m
←[94;1m== APP ==        at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
←[0m
←[94;1m== APP ==        at io.dapr.springboot.DaprApplication.start(DaprApplication.java:24)
←[0m
←[94;1m== APP ==        at io.dapr.examples.tracing.DemoService.main(DemoService.java:43)
←[0m
←[94;1m== APP ==        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
←[0m
←[94;1m== APP ==        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
←[0m
←[94;1m== APP ==        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
←[0m
←[94;1m== APP ==        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
←[0m
←[94;1m== APP ==        at io.dapr.springboot.DaprMainApplication.main(DaprMainApplication.java:29)
←[0m
←[94;1m== APP ==        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
←[0m
←[94;1m== APP ==        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
←[0m
←[94;1m== APP ==        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
←[0m
←[94;1m== APP ==        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
←[0m
←[94;1m== APP ==        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
←[0m
←[94;1m== APP ==        at org.springframework.boot.loader.Launcher.launch(Launcher.java:109)
←[0m
←[94;1m== APP ==        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
←[0m
←[94;1m== APP ==        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
←[0m
←[94;1m== APP == Caused by: org.springframework.context.annotation.ConflictingBeanDefinitionException: Annotation-specified bean name 'demoServiceController' for bean class [io.dapr.examples.tracing.DemoServiceController] conflicts with existing, non-compatible bean definition of same name and class [io.dapr.examples.invoke.http.DemoServiceController]
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.checkCandidate(ClassPathBeanDefinitionScanner.java:349)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.doScan(ClassPathBeanDefinitionScanner.java:287)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ComponentScanAnnotationParser.parse(ComponentScanAnnotationParser.java:132)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:290)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:245)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:202)
←[0m
←[94;1m== APP ==        at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:170)
←[0m
←[94;1m== APP ==        ... 25 common frames omitted

Reply:
Nice catch. Let me fix this now.